templates/simple/.envrc

```diff
@@ -1 +1,12 @@
-use flake --impure
```

Comment:
I'd put this into a separate PR, because I'd like to understand the difference and currently nix-direnv installation instructions are too confusing for mainstream adoption.

Reply:
Maybe as a middle-ground we could just source it automatically?
```
if ! has nix_direnv_version || ! nix_direnv_version 2.2.0; then
  source_url "https://raw.githubusercontent.com/nix-community/nix-direnv/2.2.0/direnvrc" "sha256-5EwyKnkJNQeXrRkYbwwRBcXbibosCJqyIUuz9Xq+LRc="
fi
```

The speed benefits and superior caching of nix-direnv are IMO worth it, without it you get issues like:
https://github.com/cachix/devenv/issues/133

Reply:
Nice, that works!

Reply:
Looks like this in practice:
```
$ cd realworld-v1-rust-actix-web-diesel 
direnv: loading ~/projects/realworld-v1-rust-actix-web-diesel/.envrc
nix-direnv: error current version v2.1.0 is older than the desired version v2.2.0
direnv: loading https://raw.githubusercontent.com/nix-community/nix-direnv/2.2.0/direnvrc (sha256-5EwyKnkJNQeXrRkYbwwRBcXbibosCJqyIUuz9Xq+LRc=)
direnv: using flake . --impure
direnv: nix-direnv: using cached dev shell
pre-commit-hooks.nix: hooks up to date
direnv: export +APIURL +C_INCLUDE_PATH +DATABASE_URL +DEVENV_DOTFILE +DEVENV_PROFILE +DEVENV_ROOT +DEVENV_STATE +FRONTEND_ORIGIN +IN_NIX_SHELL +LD_LIBRARY_PATH +LIBRARY_PATH +PC_HTTP_PORT +PC_TUI_ENABLED +PGDATA +PGHOST +PGPORT +PKG_CONFIG_PATH +name ~PATH ~XDG_CONFIG_DIRS ~XDG_DATA_DIRS
```